### PR TITLE
event util

### LINKED ISSUE #61: Event parsing function
Draft provided by @Lumyo 

```typescript
import { ContractReceipt } from 'ethers';
import { Interface } from 'ethers/lib/utils';
import { LogDescription } from '@ethersproject/abi';

function getEvents(
  contractInterface: Interface,
  receipt: ContractReceipt,
) {
  const events: LogDescription[] = [];

  for (const log of receipt.logs) {
    try {
      events.push(contractInterface.parseLog(log));
    } catch (e) {}
  }

  return events;
}
```

### PR DESCRIPTION
**Pull request by 0xCourtney**
closes #61

### CODE REVIEW

File: lib/event.ts
Comment (ItsNickBarry) [inline lib/event.ts:8]: Might as well add `except` here to continue the only/except pattern.  But I'm thinking about making a separate package to handle that logic because it's used it so many places, so we can just wait for that.

File: lib/event.ts
Comment (ItsNickBarry) [inline lib/event.ts:14]:
```suggestion
      const description = contractInterface.parseLog(log);
```

File: lib/event.ts
Comment (ItsNickBarry) [inline lib/event.ts:20]: Already parsed, so just push `description`.
```suggestion
      events.push(description);
```

**Review by ItsNickBarry — CHANGES_REQUESTED** [review]

**Comment by ItsNickBarry** [comment]
@0xCourtney Is this something you want published now, or are you just trying to resolve the open issue?

**Comment by 0xCourtney** [comment]
> @0xCourtney Is this something you want published now, or are you just trying to resolve the open issue?

This is not urgent. I was thinking about this a bit recently and thought it would be good to share what I worked on.

File: lib/event.ts
Comment (0xCourtney) [reply lib/event.ts:8]: Absolutely, the filter I did for Diamond utils also supports `except` so it may make the most sense to wait until that's merged or branch from that PR and break out the filters specifically.

File: lib/event.ts
Comment (ItsNickBarry) [inline lib/event.ts:14]: I don't like this variable name.  Can change it in a future reorganization, though.

**Review by ItsNickBarry — APPROVED** [review]

File: lib/event.ts
Comment (0xCourtney) [reply lib/event.ts]: renamed this to `parsedLog`

**Review by ItsNickBarry — APPROVED** [review]
Merge if ready.

**Pull request merged**
